apps/aecore/lib/aecore/chain/identifier.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Aecore.Chain.Identifier do
+  alias __MODULE__
+
+  @moduledoc """
+  Utility module for interacting with identifiers. 
+
+  Our binaries like account pubkey or hashes will already be represented as encoded (with already specified tag) binaries, following  the next formula:
+   <<Tag:1/unsigned-integer-unit:8, Binary:32/binary-unit:8>>, Where 
+   Tag - is  a non-negative integer , a number from a range of 1..6 (at the current state of this documentation, for more info get in config.exs file and find :aecore, :binary_ids list), 
+   Binary - is a regular binary , which byte size is 32 bytes.
+  Example:
+  Epoch has a separate structure for id's:
+  %Identifier{tag: atom(), value: value() }
+  Account structure:
+
+   In Epoch's erlang core implementation (as it would look like the following Elixir code):
+   %Account{
+    pubkey: %Identifier{tag: :account, value: <<"some_pub_key">>},
+    nonce: non_neg_integer(),
+    balance: non_neg_integer()
+  }
+  """
+  @type t() :: %Identifier{type: type(), value: value()}
+  @type type() :: :account | :name | :commitment | :oracle | :contract | :channel
+  # byte_size should be 32 byte
+  @type value() :: binary()
+  defstruct type: :undefined, value: ""
+  use ExConstructor
+
+  @spec create_identity(type(), value()) :: Identifier.t() | {:error, String.t()}
+  # byte_size(data) == 32 data should be stricted to 32 bytes only
+  def create_identity(value, type) when is_atom(type) and is_binary(value) do
+    case Application.get_env(:aecore, :binary_ids)[type] do
+      nil ->
+        {:error,
+         "#{__MODULE__}: The following tag: #{inspect(type)} for given value: #{inspect(value)} doesn't exist"}
+
+      tag when is_integer(tag) ->
+        {:ok, %Identifier{type: type, value: value}}
+
+      _ ->
+        create_identity(value, type)
```

Comment:
Wouldn't this cause infinite recursion? (We call ourselves with the same arguments so this case will by lunched again, and again...) I think we should "error" here instead.

Reply:
I will adjust it.